R/dummy_extract.R

```diff
@@ -0,0 +1,280 @@
+#' Regular Expression Dummy Variables Creation
+#'
+#' `step_dummy_extract()` creates a *specification* of a recipe
+#'  step that will convert nominal data (e.g. character or factors)
+#'  into one or more integer model terms for the extracted levels.
+#'
+#' @inheritParams step_center
+#' @inheritParams step_other
+#' @inheritParams step_dummy
+#' @param sep Character vector containing a regular expression to use
+#'   for splitting. [strsplit()] is used to perform the split. `sep` takes
+#'   priority if `pattern` is also specified.
+#' @param pattern Character vector containing a regular expression used
+#'   for extraction. [gregexpr()] and [regmatches()] are used to perform
+#'   pattern extraction using `perl = TRUE`.
+#' @template step-return
+#' @family dummy variable and encoding steps
+#' @seealso [dummy_names()]
+#' @export
+#' @details `step_dummy_extract()` will create a set of integer dummy
+#'  variables from a character variable by extract individual strings
+#'  by either splitting or extracting then counting those to create
+#'  count variables.
+#'
+#'  Note that `threshold` works in a very specific way for this step.
```

Comment:
About `threshold`, I think that it should default to zero for this step (and changed it to zero). We shouldn't remove any by default (as opposed to `step_other()` where filtering is all that the step does).